Sources/Hedera/MirrorNodeService.swift

```diff
@@ -0,0 +1,208 @@
+/*
+ * ‌
+ * Hedera Swift SDK
+ *
+ * Copyright (C) 2022 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import AsyncHTTPClient
+import Foundation
+import HederaProtobufs
+
+internal final class MirrorNodeService {
+    internal var mirrorNodeGateway: MirrorNodeGateway
+
+    init(_ mirrorNodeGateway: MirrorNodeGateway) {
+        self.mirrorNodeGateway = mirrorNodeGateway
+    }
+
+    internal func getAccountNum(_ evmAddress: String) async throws -> UInt64 {
+        let accountInfoResponse = try await self.mirrorNodeGateway.getAccountInfo(evmAddress)
+
+        guard let accountId = accountInfoResponse["account"] else {
+            throw NSError(
+                domain: "InvalidResponseError", code: -1,
+                userInfo: [NSLocalizedDescriptionKey: "Error while processing getAccountInfo mirror node query"])
+        }
```

Comment:
following above with `Codable`, this should be a `struct AccountInfoResponse: Codable`, makes the parsing a lot easier.